files/task_helper.rb

```diff
@@ -46,7 +46,13 @@ def self.run
     # the task. Unhandled errors are caught and turned into an error result.
     # @param [Hash] params A hash of params for the task
     # @return [Hash] The result of the task
-    result = new.task(params)
+
+    result = if params.dig(:_target, :protocol) &&
```

Comment:
the protocol of a remote target is not necessarily `remote`.  the `_target` metaparam exists solely for remote targets which may use other protocols.

`dig` and `==` is redundant too